lib/rules/async-server-action.js

```diff
@@ -0,0 +1,52 @@
+/**
+ * @fileoverview Require functions with the `use server` directive to be async
+ * @author Jorge Zreik
+ */
+
+'use strict';
+
+const docsUrl = require('../util/docsUrl');
+const report = require('../util/report');
+
+// ------------------------------------------------------------------------------
+// Rule Definition
+// ------------------------------------------------------------------------------
+
+const messages = {
+  asyncServerAction: 'Your server action should be async',
+};
+
+module.exports = {
+  meta: {
+    docs: {
+      description: 'Require functions with the `use server` directive to be async',
+      category: 'Possible Errors',
+      recommended: false,
+      url: docsUrl('async-server-action'),
+    },
+
+    messages,
+
+    fixable: 'code',
+
+    schema: [],
+  },
+
+  create(context) {
+    let currentFunction = null;
+
+    return {
+      ':function'(node) {
+        currentFunction = node;
+      },
+      ':function[async=false]>BlockStatement>:first-child[expression.value="use server"]'() {
+        report(context, messages.asyncServerAction, 'asyncServerAction', {
+          node: currentFunction,
+          fix(fixer) {
+            return fixer.insertTextBefore(currentFunction, 'async ');
+          },
```

Comment:
i'm not 100% convinced this should be an autofix (as opposed to a suggestion). If there is _any_ situation where the proper fix is to remove the "use server" pragma, instead of adding `async`, then an autofix wouldn't be appropriate.

Reply:
After thinking about it for a while I do think a suggestion is better. 99% of the time it should be better to add async, but I can imagine edge cases. Also a suggestion would do a better job of informing why async is required.

Reply:
Changed to suggest.